v2/api/dataprotection/v1api20230101/structure.txt

```diff
@@ -0,0 +1,1272 @@
+│   │   │   │   ├── ResourceType: *string
+│   │   │   │   └── ResourceUri: *string
+│   │   │   ├── DataSourceSetInfo: *Object (7 properties)
+│   │   │   │   ├── DatasourceType: *string
+│   │   │   │   ├── ObjectType: *string
+│   │   │   │   ├── ResourceLocation: *string
+│   │   │   │   ├── ResourceName: *string
+│   │   │   │   ├── ResourceReference: *genruntime.ResourceReference
+│   │   │   │   ├── ResourceType: *string
+│   │   │   │   └── ResourceUri: *string
+│   │   │   ├── DatasourceAuthCredentials: *Object (1 property)
+│   │   │   │   └── SecretStoreBasedAuthCredentials: *Object (2 properties)
+│   │   │   │       ├── ObjectType: *Enum (1 value)
+│   │   │   │       │   └── "SecretStoreBasedAuthCredentials"
+│   │   │   │       └── SecretStoreResource: *Object (3 properties)
+│   │   │   │           ├── SecretStoreType: *Enum (2 values)
+│   │   │   │           │   ├── "AzureKeyVault"
+│   │   │   │           │   └── "Invalid"
+│   │   │   │           ├── Uri: *string
+│   │   │   │           └── Value: *string
+│   │   │   ├── FriendlyName: *string
+│   │   │   ├── ObjectType: *string
+│   │   │   ├── PolicyInfo: *Object (2 properties)
+│   │   │   │   ├── PolicyId: *string
+│   │   │   │   └── PolicyParameters: *Object (2 properties)
+│   │   │   │       ├── BackupDatasourceParametersList: Object (2 properties)[]
+│   │   │   │       │   ├── Blob: *Object (2 properties)
+│   │   │   │       │   │   ├── ContainersList: string[]
+│   │   │   │       │   │   └── ObjectType: *Enum (1 value)
+│   │   │   │       │   │       └── "BlobBackupDatasourceParameters"
+│   │   │   │       │   └── KubernetesCluster: *Object (8 properties)
+│   │   │   │       │       ├── ExcludedNamespaces: string[]
+│   │   │   │       │       ├── ExcludedResourceTypes: string[]
+│   │   │   │       │       ├── IncludeClusterScopeResources: *bool
+│   │   │   │       │       ├── IncludedNamespaces: string[]
+│   │   │   │       │       ├── IncludedResourceTypes: string[]
+│   │   │   │       │       ├── LabelSelectors: string[]
+│   │   │   │       │       ├── ObjectType: *Enum (1 value)
+│   │   │   │       │       │   └── "KubernetesClusterBackupDatasourceParameters"
+│   │   │   │       │       └── SnapshotVolumes: *bool
+│   │   │   │       └── DataStoreParametersList: Object (1 property)[]
+│   │   │   │           └── AzureOperationalStoreParameters: *Object (3 properties)
+│   │   │   │               ├── DataStoreType: *Enum (3 values)
+│   │   │   │               │   ├── "ArchiveStore"
+│   │   │   │               │   ├── "OperationalStore"
+│   │   │   │               │   └── "VaultStore"
+│   │   │   │               ├── ObjectType: *Enum (1 value)
+│   │   │   │               │   └── "AzureOperationalStoreParameters"
+│   │   │   │               └── ResourceGroupId: *string
+│   │   │   └── ValidationType: *Enum (2 values)
+│   │   │       ├── "DeepValidation"
+│   │   │       └── "ShallowValidation"
+│   │   ├── SystemData: *Object (6 properties)
```

Comment:
Should this be marked readonly, like you did for the other resource? (see your existing edits to azure-arm.yaml)